muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -272,6 +272,58 @@ public Response retrieveForm(){
         vopsController.loadStringProperty(user, "religion"),
         vopsController.loadStringProperty(user, "additionalInfo")
     );
+  }
+
+  @GET
+  @Path("/hops")
+  @RESTPermit(handling=Handling.INLINE)
+  public Response retrieveForm(){
+
+    if (!sessionController.isLoggedIn()) {
+      return Response.status(Status.FORBIDDEN).entity("Must be logged in").build();
+    }
+
+    SchoolDataIdentifier userIdentifier = sessionController.getLoggedUser();
+    
+    HopsRESTModel response = createHopsRESTModelForStudent(userIdentifier);
+    
+    if (response == null) {
+      return Response.status(Status.NOT_FOUND).entity("No HOPS form for non-students").build();
+    }
+    
+    return Response.ok(response).build();
+  }
+
+  @GET
+  @Path("/hops/{USERIDENTIFIER}")
+  @RESTPermit(handling=Handling.INLINE)
+  public Response retrieveForm(@PathParam("USERIDENTIFIER") String userIdentifierString){
+    
+    // TODO security
```

Comment:
Do do. (I guess the same check applies that is above (logged user for himself / permission for any.)